spring-vault-core/src/main/java/org/springframework/vault/authentication/AzureVmEnvironment.java

```diff
@@ -37,21 +37,26 @@
 
 	private final String vmName;
 
+	private final String vmScaleSetName;
+
 	/**
 	 * Creates a new {@link AzureVmEnvironment}.
 	 * @param subscriptionId must not be {@literal null}.
 	 * @param resourceGroupName must not be {@literal null}.
 	 * @param vmName must not be {@literal null}.
+	 * @param vmScaleSetName must not be {@literal null}.
 	 */
-	public AzureVmEnvironment(String subscriptionId, String resourceGroupName, String vmName) {
+	public AzureVmEnvironment(String subscriptionId, String resourceGroupName, String vmName, String vmScaleSetName) {
```

Comment:
Care to add another constructor to retain binary compatibility? We should have both constructors, one with and one without `vmScaleSet`.

Reply:
Done